src/Reflection/ReflectionParameter.php

```diff
@@ -271,7 +271,10 @@ public function isDefaultValueAvailable(): bool
      */
     public function getDefaultValue(): string|int|float|bool|array|null
     {
-        return $this->getCompiledDefaultValue()->value;
+        /** @psalm-var scalar|array<scalar>|null $value */
```

Comment:
Also here: default values could perhaps be objects? 🤔

Reply:
https://github.com/Roave/BetterReflection/blob/5.1.x/src/NodeCompiler/CompileNodeToValue.php#L64